e2e/libs/keywords/volume_keywords.py

```diff
@@ -33,26 +49,57 @@ def wait_for_volume_expand_to_size(self, volume_name, size):
         logging(f'Waiting for volume {volume_name} expand to {size}')
         return self.volume.wait_for_volume_expand_to_size(volume_name, size)
 
-    def get_volume_node(self, volume_name):
+    def get_replica_nodes(self, volume_name):
+        self.get_replica_node_attached_to_volume(volume_name)
+        self.get_replica_node_not_attached_to_volume(volume_name)
+
+    def get_replica_node_attached_to_volume(self, volume_name):
         volume = self.volume.get(volume_name)
+        ResourceCache.create_or_update_link(
+            KIND_VOLUME, volume_name, CACHE_NODE_ID_OF_REPLICA_0, volume['spec']['nodeID']
+        )
         return volume['spec']['nodeID']
 
-    def get_replica_node(self, volume_name):
+    def get_replica_node_not_attached_to_volume(self, volume_name):
         worker_nodes = list_node_names_by_role("worker")
-        volume_node = self.get_volume_node(volume_name)
+        volume_node = self.get_replica_node_attached_to_volume(volume_name)
+        replica_nodes = [node for node in worker_nodes if node != volume_node]
+
         test_pod_running_node = get_test_pod_running_node()
-        for worker_node in worker_nodes:
-            if worker_node != volume_node and worker_node != test_pod_running_node:
-                return worker_node
+
+        replica_1 = None
+        replica_2 = None
+
+        for replica_node in replica_nodes:
+            if replica_node == test_pod_running_node:
+                replica_2 = replica_node
+                continue
+
+            if replica_1 is None:
+                replica_1 = replica_node
+                continue
+            else:
+                replica_2 = replica_node
+                break
+
+        ResourceCache.create_or_update_link(
+            KIND_VOLUME, volume_name, CACHE_NODE_ID_OF_REPLICA_1, replica_1
+        )
+        ResourceCache.create_or_update_link(
+            KIND_VOLUME, volume_name, CACHE_NODE_ID_OF_REPLICA_2, replica_2
+        )
```

Comment:
Why we just hard-code and limit the number of replica nodes only 2? We should not make this hard assumption.

We can just cache the created resources. For the relationship, they seem not really needed, as we can query them at runtime. For testing, we don't need to worry about performance issue.